### ISSUE TITLE
Is it possible to remove AU3_ from the functions

### ISSUE THREAD

**Issue by rahulmr** [midorlo/JNAutoIt#1]
I like the implementation which does not need jacob.jar and DLLs but https://github.com/accessrichard/autoitx4java implementation has better names.
Could you please suggest how this can be implemented?

For example instead of `aix.AU3_MouseClick();` can we just have `aix.mouseClick();`

**Comment by midorlo** [comment]
Yes it's possible, but it would break its current usage in other projects.
Maybe just wrap the object?

Am Di., 22. Jan. 2019, 13:40 hat Rahul Raut <notifications@github.com>
geschrieben:

**Comment by rahulmr** [comment]
@midorlo actually I tried but it seems it will take a lot of coding. Actually, I want to use Rhino JS to work with AutoIT functions but in javascript HWND, LPWSTR, etc are not there so I wanted functions to have only string , int or float as parameters. Not sure how to do this.
Also, there is no exception shown when we use the library. Like if it doesn't get the required window it is not throwing error. It simply executes without doing anything.

**Comment by midorlo** [comment]
Unfortunately that's the nature of AutoIt. It language does not have try-catch but stores the current error state in `@error` 

As I didn't write any redesigns but just implemented their API in Java. You possibly need to check [AU3_error()](https://github.com/midorlo/JNAutoIt/blob/22b369dde5f4efa3cf805c98a479faa006733a20/src/main/java/de/midorlo/jnautoit/jna/AutoItX.java#L202) from your Singleton.

**Comment by rahulmr** [comment]
Thanks @midorlo !! Will try that.
Just one last question, after which you can close the issue, are you sure that AutoItX dll is not required to be registered?

**Comment by midorlo** [comment]
x86/amd64 versions of the dll are bundled in the project and get provided using JNA on runtime (when constructing an Instance of `AutoItX`:

`(AutoItXLibrary) Native.loadLibrary(DLL, AutoItXLibrary.class);`

That said, my understandings of JNA next to non-existant. If the code is wrong, I'd love to know about it and why.

**Comment by rahulmr** [comment]
@midorlo I loved your work and I am able to integrate it will Sahi Pro web automation tool. Also, I was able to pack the dll files in the jar so I don't have to keep them in some location on the path.
Thanks.

**Comment by midorlo** [comment]
Thank you :)

Still, it wonders me that you bundled the dll yourself. They _are_ bundled in this project ([see ](https://github.com/midorlo/JNAutoIt/tree/master/src/main/resources) `/src/main/resources`) and should therefore be available to `AutoItX` at any time.
